### PR TITLE
Add Type Hints as specified in is-charms-contributing-guide

### PR DESCRIPTION
**Pull request by javierdelapuente**
Applicable spec: <link>

### Overview

Add the type hints that are specified in the [is-charms-contributing-guide](https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#type-hints).

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by arturo-seijas — DISMISSED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -92,7 +94,7 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent):
             self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
 
     @property
-    def _pebble_layer(self):
+    def _pebble_layer(self) -> typing.Union[pebble.LayerDict]:
```
Comment (cbartz) [inline src/charm.py:97]:
```suggestion
    def _pebble_layer(self) -> pebble.LayerDict:
```
What's the point of having Union with one type here?

Comment (javierdelapuente) [reply src/charm.py]: No reason at all... Thanks!

**Review by arturo-seijas — DISMISSED** [review]

**Review by cbartz — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

**Review by merkata — APPROVED** [review]

File: .github/workflows/integration_test.yaml [hunk 2]
```diff
@@ -8,9 +8,6 @@ jobs:
     uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
     secrets: inherit
     with:
-      chaos-app-label: app.kubernetes.io/name=indico
-      chaos-enabled: false
-      chaos-experiments: pod-delete
```
Comment (cbartz) [inline .github/workflows/integration_test.yaml:13]: @javierdelapuente  Could you please add a small note to the PR description stating that the chaos cleanup is also done (as this is unrelated to the original intent of the PR)?Thanks.
